jerry-core/api/jerry-snapshot.c

```diff
@@ -580,34 +584,32 @@ jerry_exec_snapshot (const uint32_t *snapshot_p, /**< snapshot */
     return ecma_raise_type_error (invalid_version_error_p);
   }
 
-  lit_mem_to_snapshot_id_map_entry_t *lit_map_p = NULL;
-  uint32_t literals_num;
-
   if (header_p->lit_table_offset >= snapshot_size)
   {
     return ecma_raise_type_error (invalid_version_error_p);
   }
 
-  JERRY_ASSERT ((header_p->lit_table_offset % sizeof (uint32_t)) == 0);
-  if (!ecma_load_literals_from_snapshot ((uint32_t *) (snapshot_data_p + header_p->lit_table_offset),
-                                         header_p->lit_table_size,
-                                         &lit_map_p,
-                                         &literals_num))
+  if (func_index >= header_p->number_of_funcs)
   {
-    JERRY_ASSERT (lit_map_p == NULL);
-    return ecma_raise_type_error (invalid_format_error_p);
+    return ecma_raise_type_error ("Function index is higher than maximum");
```

Comment:
would be better (smaller) to have error messages the jerry.c way:
```c
#ifdef JERRY_ENABLE_ERROR_MESSAGES
static const char * const some_msg_p = "some error message";
#endif /* JERRY_ENABLE_ERROR_MESSAGES */

jerry_value_t
some_function ()
{
  return ecma_raise_type_error (ECMA_ERR_MSG (some_msg_p));
}
```
Then, if error messages are disabled, the string literal does not get built in the binary.

(Unfortunately, this is only used in jerry.c consistently.)